hypothesis-python/tests/conjecture/common.py

```diff
@@ -67,3 +71,142 @@ def accept(f):
             )
 
     return accept
+
+
+def fresh_data():
+    return ConjectureData(BUFFER_SIZE, prefix=b"", random=Random())
+
+
+@st.composite
+def draw_integer_kwargs(
+    draw,
+    *,
+    use_min_value=True,
+    use_max_value=True,
+    use_shrink_towards=True,
+    use_weights=True,
+    use_forced=False,
+):
+    min_value = None
+    max_value = None
+    shrink_towards = 0
+    weights = None
+
+    forced = draw(st.integers()) if use_forced else None
+    if use_min_value:
+        min_value = draw(st.integers(max_value=forced))
+    if use_max_value:
+        min_vals = []
+        if min_value is not None:
+            min_vals.append(min_value)
+        if forced is not None:
+            min_vals.append(forced)
+        min_val = max(min_vals) if min_vals else None
+        max_value = draw(st.integers(min_value=min_val))
```

Comment:
```suggestion
        max_val = min_val + 1023 if use_weights else None
        max_value = draw(st.integers(min_value=min_val))
```
I think this'll make the use_weights case considerably more efficient.  

Better still would be to draw the weights list, and then derive a max_value from the length of that list - "draw integer and then that many elements" is a classic tarpit for our engine, and 1024 floats is a large fraction of our available entropy.

Reply:
It's a bit more complicated than this, because we have two invariants to maintain, either of which may be relaxed if the relevant parameter is not specified (`min_value <= forced <= max_value`, and `max_value - min_value + 1 == len(weights)`).

You've nerd sniped me though, so I pushed an improvement here. Hopefully it's also readable ish.